src/genericmemory.c

```diff
@@ -193,6 +193,7 @@ JL_DLLEXPORT jl_value_t *jl_genericmemory_to_string(jl_genericmemory_t *m, size_
     }
     int how = jl_genericmemory_how(m);
     size_t mlength = m->length;
+    jl_gc_count_freed(mlength);
```

Comment:
this should go later, as right now you are double-counting the free in a couple cases